frontend/src/metabase/nav/components/SearchBar.unit.spec.tsx

```diff
@@ -0,0 +1,182 @@
+  return screen.getByPlaceholderText("Search…");
+};
+
+describe("SearchBar", () => {
+  describe("typing a search query", () => {
+    it("should change URL when user types a query and hits `Enter`", async () => {
+      const { history } = setup();
+
+      userEvent.type(getSearchBar(), "BC{enter}");
+
+      const location = history?.getCurrentLocation();
+      expect(location?.pathname).toEqual("search");
+      expect(location?.search).toEqual("?q=BC");
+    });
+    it("should render 'No Results Found' when the query has no results", async () => {
+      setup({ searchResultItems: [] });
+      const searchBar = getSearchBar();
+      userEvent.click(searchBar);
+      userEvent.type(searchBar, "XXXXX");
+      await waitFor(() =>
+        expect(screen.queryByTestId("loading-spinner")).not.toBeInTheDocument(),
+      );
+
+      expect(
+        within(screen.getByTestId("search-bar-results-container")).getByText(
+          "Didn't find anything",
+        ),
+      ).toBeInTheDocument();
+    });
+  });
+  describe("focusing on search bar", () => {
+    it("should render `Recent Searches` list when clicking the search bar", async () => {
+      setup();
+      getSearchBar().click();
+      expect(await screen.findByText("Recents ABC")).toBeInTheDocument();
+    });
+
+    it("should render `Nothing here` and a folder icon if there are no recently viewed items", async () => {
+      setup({ recentViewsItems: [] });
+      getSearchBar().click();
+      expect(await screen.findByText("Nothing here")).toBeInTheDocument();
+    });
+  });
+  describe("keyboard navigation", () => {
+    it("should focus on the filter bar when the user tabs from the search bar", () => {
+      setup();
+      getSearchBar().click();
+      userEvent.tab();
+      expect(screen.getByTestId("search-bar-filter-button")).toHaveFocus();
+    });
+    it("should allow navigation through the search results with the keyboard", async () => {
+      setup();
+      getSearchBar().click();
+      userEvent.type(getSearchBar(), "BC");
+
+      await waitFor(() =>
+        expect(screen.queryByTestId("loading-spinner")).not.toBeInTheDocument(),
+      );
+
+      const resultItems = await screen.findAllByTestId("search-result-item");
+      expect(resultItems.length).toBe(2);
+
+      // tab over the filter button
+      userEvent.tab();
+
+      // There are two search results, each with a link to `Our analytics`,
+      // so we want to navigate to the search result, then the collection link.
+      for (const substring of ["Card ABC", "Card BCD"]) {
+        userEvent.tab();
+
+        const filteredElement = resultItems.find(element =>
+          element.textContent?.includes(substring),
+        );
+
+        expect(filteredElement).not.toBeUndefined();
+        expect(filteredElement).toHaveFocus();
+
+        userEvent.tab();
+
+        expect(
+          within(filteredElement as HTMLElement).getByText("Our analytics"),
+        ).toHaveFocus();
```

Comment:
Am I correct that:
1. first tab: `Card ABC` is focused
2. 2nd tab: `Our analytics` is focused
3. 3rd tab: `Card BCD` is focused
4. 4th tab: `Our analytics` is focused

I kind of expect it'd look something like
1. first tab: `Card ABC` is focused
1. 2nd tab: `Card BCD` is focused
1. 3rd tab: `Our analytics` is focused

I think we assert `Our analytics` focused twice since it's in the for loop. Can you include screenshots of how this is working?

Reply:
![Screenshot 2023-08-03 at 2 12 26 PM 1](https://github.com/metabase/metabase/assets/25306947/95a5eb39-b739-4f9b-969f-43ffe6dc70ca)

Basically we have a link to the parent collection for each search element - so for this test `Our analytics` is the parent element rather than another entry within the search. So when we're tabbing through, we'll want to tab into the search result, then tab to the parent collection link, then tab to the next search result, and so on

Reply:
Thanks for the explanation. I thought our analytics would be the 3rd element or something.